code/obj/machinery/sleeperscanner.dm

```diff
@@ -0,0 +1,28 @@
+/obj/machinery/sleeperscanner
+	name = "Hand Scanner"
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "handscanner"
+	desc = "A device embedded into the wall, it has a image of a handprint on it."
+
+/obj/machinery/sleeperscanner/attackby(obj/item/W, mob/user as mob)
+	if(istype(W, /obj/item/device/detective_scanner))
+		return
+	if (user.mind?.get_antagonist(ROLE_SLEEPER_AGENT))
+		user.visible_message("<span class='notice'>The [src] accepts the biometrics of the user and beeps, granting you access.</span>")
+		for (var/obj/machinery/door/airlock/pyro/reinforced/syndicate/M in by_type[/obj/machinery/door])
```

Comment:
The type `/obj/machinery/door/airlock/pyro/reinforced/syndicate` is used in several prefabs and in Z2; listening post doors will require to be their own subtype if this is to work.
I would personally recommend refactoring sleeper hand scanners into a subtype of `/obj/machinery/door_control`, and utilising `id` to open the listening post doors.